grpc/doc/swdesign/README.md

```diff
@@ -45,13 +51,190 @@ Considered alternatives:
 * implementing the gRPC protocol completely inside the middleware
 * using an http library and a protobuf converter
 
+#### Support PEM file format for X509 certificates
+`swdd~grpc-supports-pem-file-format-for-X509-certificates~1`
+
+Status: approved
+
+When a PEM file is provided as certificate, the gRPC middleware shall:
+* parse the given PEM file into x509 certificate buffer format.
+* use the parsed x509 certificate to setup the TLS configuration.
+
+Rationale:
+
+The advantage of using a PEM file is due to its text-based, human-readable format, making it more versatile, as it can contain certificates, private keys, public keys and even certificate chain, compared to DER.
+
+Needs:
+- impl
+- utest
+
+Assumptions:
+
+No assumptions were taken.
+
+#### Support PEM file format for keys
+`swdd~grpc-supports-pem-file-format-for-keys~1`
+
+Status: approved
+
+When a PEM file is provided as a key, the gRPC middleware shall:
+* parse the given PEM file into a buffer containing the key.
+* use the parsed key to setup the TLS configuration.
+
+Rationale:
+
+The advantage of using a PEM file is due to its text-based, human-readable format, making it more versatile, as it can contain certificates, private keys, public keys and even certificate chain, compared to DER.
+
+Needs:
+- impl
+- utest
+
+Assumptions:
+
+No assumptions were taken.
+
+#### Check given PEM file for proper unix file permission
+`swdd~grpc-checks-given-PEM-file-for-proper-unix-file-permission~1`
+
+Status: approved
+
+When a PEM file is provided, the gRPC middleware shall check the given PEM file to have one of the following unix file permission:
+* 400
+* 600
+
+Rationale:
+
+Due to security reasons certificates and keys shall not be read- and writable by groups and others.
+
+Needs:
+- impl
+- utest
+
+Assumptions:
+
+No assumptions were taken.
+
+#### Activate mTLS on gRPC server when certificates and key are provided upon the start
+`swdd~grpc-server-activate-mtls-when-certificates-and-key-provided-upon-start~1`
+
+Status: approved
+
+When the root, the server's certificate and the key is provided upon start of the gRPC server, the gRPC server shall use the provided certificates and the key to activate mTLS for the gRPC communication between the involved parties.
+
+Rationale:
+
+To avoid complexity, coming with mTLS configuration e.g. certificates generation and management, during development phase, mTLS can be activated on demand.
```

Comment:
Is it necessary to write it with every requirement? What about writing this rationale to the SwDD as design decision (and remove it from here)?

Reply:
Done.